java/core/src/main/java/co/worklytics/psoxy/gateway/output/ApiDataOutputUtils.java

```diff
@@ -112,8 +115,11 @@ public ProcessedContent responseAsRawProcessedContent(HttpRequest sourceApiReque
             .contentType(sourceApiResponse.getContentType())
             .contentCharset(sourceApiResponse.getContentCharset());
 
-        try (InputStream stream = sourceApiResponse.getContent()) {
-            builder.content(stream.readAllBytes());
+        // sourceApiResponse will not have content for 'HEAD' requests (or potentially even GET, in some cases - 204 No Content, 304 Not Modified, etc.)
+        if (sourceApiResponse.getContent() != null) {
```

Comment:
OK;
I think is missing to support 304 as a successful response:

https://github.com/Worklytics/psoxy/blob/926bdf16be39a6b128dc6fa613ca66569dd39d70/java/core/src/main/java/co/worklytics/psoxy/gateway/impl/ApiDataRequestHandler.java#L698

As currently in this case, handler will return an error due response is < 300

Reply:
the 304 this in in general; the `isSuccessFamily` thing - we have to look a how that is used. For our use-case, I think correct.

the better question is that if we're using `Get-If-Modified-Since`, which could result in the 304 - then we should consider if we want some other method than `isSuccessFamily`.

afaik, proxy passes non-200s back to caller anyways.

Reply:
Yes, but when  *non success* it includes the api error header in the response:

https://github.com/Worklytics/psoxy/blob/926bdf16be39a6b128dc6fa613ca66569dd39d70/java/core/src/main/java/co/worklytics/psoxy/gateway/impl/ApiDataRequestHandler.java#L480

We are not currently using 304 (it was a test that I was doing yesterday for a thing that we are not going to need ATM)  and its headers (like `Get-If-Modified-Since`) will currently need to be part of the rules as by default they are omitted when doing the call